cytoscape-matterjs.js

```diff
@@ -0,0 +1,428 @@
+      var matterNodes = [];
+      var matterEdges = [];
+      var matterCenterSprings = [{ id: 'top', children: [] }];
+
+// +------------------------------------------------------------------------+ //
+// +-----------------ADDS NODES TO THE MATTER JS SIMULATION-----------------+ //
+// +------------------------------------------------------------------------+ //
+// +------------------------------------------------------------------------+ //
+
+      function addToCenterSprings(i) {
+        if (matterNodes[i].parent !== undefined) {
+          for (var j = 0; j < matterCenterSprings.length; j++) {
+            if (matterCenterSprings[j].id === matterNodes[i].parent) {
+              matterCenterSprings[j].children.push({ id: matterNodes[i].shape.id, x: matterNodes[i].shape.position.x, y: matterNodes[i].shape.position.y });
+            }
+          }
+        } else {
+          for (var j = 0; j < matterCenterSprings.length; j++) {
+            if (matterCenterSprings[j].id === 'top') {
+              matterCenterSprings[j].children.push({ id: matterNodes[i].shape.id, x: matterNodes[i].shape.position.x, y: matterNodes[i].shape.position.y });
+            }
+          }
+        }
+      }
+
+      function mapNodes(nodesIn) {
+        var currentMatterNode = 0;
+        for (var i = 0; i < nodesIn.length; i++) {
+          if (nodesIn[i].parent().length === 0) {
+            if (nodesIn[i].children().length > 0) {
+
+              var mass = 10;
+              for (var j = 0; j < options.mass.length; j++) {
+                if (options.mass[j].id === nodesIn[i].id()) {
+                  mass = options.mass[j].mass;
+                  console.log(options.mass[j].id);
+                }
+              }
+
+              matterNodes.push({
+                shape: Bodies.rectangle(
+                  nodesIn[i].position().x,
+                  nodesIn[i].position().y,
+                  nodesIn[i].width(),
+                  nodesIn[i].height(),
+                  {
+                    id: nodesIn[i].data().id,
+                    inertia: Infinity,
+                    mass: mass,
+                    gravity: options.gravity,
+                    frictionAir: options.globalAirFriction,
+                    render: {
+                      strokeStyle: Common.shadeColor('#4ECDC4', -20),
+                      fillStyle: '#4ECDC4',
+                    },
+                  }
+                ),
+                parent: nodesIn[i].data().parent,
+                children: [],
+                oldX: nodesIn[i].position().x,
+                oldY: nodesIn[i].position().y
+              });
+
+              World.addBody(engine.world, matterNodes[currentMatterNode].shape);
+              matterCenterSprings.push({ id: matterNodes[currentMatterNode].shape.id, children: [] });
+
+              addToCenterSprings(currentMatterNode);
+            } else {
+              matterNodes.push({
+                shape:Bodies.circle(
```

Comment:
It might be better to use a rect since many node shapes can use the full rect/bb